### ISSUE TITLE
"Cast target is not a type" false positive on tuple literal types

### ISSUE THREAD

**Issue by DetachHead** [KotlinIsland/basedmypy#502]
### Describe the problem, ie expected/actual result (if it's not blatantly obvious)

this works at runtime so it should be allowed

### Gist to reproduce

```python
from typing import cast

cast((int, str), 1) # error: Cast target is not a type  [misc]
```

**Comment by KotlinIsland** [comment]
merged with #378
